lang/cs/Org.Apache.REEF.Network/NetworkService/WritableNsMessage.cs

```diff
@@ -144,12 +146,13 @@ public void Write(IDataWriter writer)
             SourceId = _factory.Create(await reader.ReadStringAsync(token));
             DestId = _factory.Create(await reader.ReadStringAsync(token));
             int messageCount = await reader.ReadInt32Async(token);
+            string dataType = await reader.ReadStringAsync(token);
 
             Data = new List<T>();
 
             for (int index = 0; index < messageCount; index++)
             {
-                var dataPoint = Activator.CreateInstance<T>();
+                var dataPoint = (T) _injection.ForkInjector().GetInstance(Type.GetType(dataType));
```

Comment:
This would be resolved when we move away from `Writable`, right? At that point, we can keep reusing the same `StreamingCodec`, correct?